src/geometry/rtree.cpp

```diff
@@ -252,32 +252,41 @@ std::size_t RTree<cs>::report() const noexcept {
 
 template <typename cs>
 template <typename Predicate, typename Filter>
-auto RTree<cs>::apply_predicate(Predicate&& predicate, Filter&& filter) const -> query_results {
-  return rtree_.qbegin(std::forward<Predicate>(predicate) &&
+auto RTree<cs>::apply_intersect_predicate(Predicate&& predicate, Filter&& filter) const -> query_results {
+  return rtree_.qbegin(predicate &&
```

Comment:
Why calling predicate outside of the lambda?
Would the following snippet have worked as well?
```
bgi::satisfies([this, p=std::forwared(predicate), f = std::forward(filter)](doc_type const& doc) -> bool {
                         return p && lookup(doc).map(f).value_or(false);
                       }
```